### ISSUE TITLE
shellcheck disable creates fatal error when chaining commands

### ISSUE THREAD

**Issue by kkmuffme** [koalaman/shellcheck#2672]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1073, SC1126, SC1072
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
echo "a" &&

# shellcheck disable=SC2016
echo "b"
        
```

#### Here's what shellcheck currently says:
```
Line 5	SC1073: Couldn't parse this simple command. Fix to allow more checks.
Line 5	SC1126: Place shellcheck directives before commands, not after.
Line 5	SC1072: Expected a command. Fix any mentioned problems and try again.
```
#### Here's what I wanted or expected to see:

No error - when chaining commands that are in separate lines, then shellcheck should correctly handle "shellcheck disable=" comments (like it already does for all other comments) and not fail.

**Comment by TinCanTech** [comment]
The way `shellcheck` expect this to be written is:
```
# shellcheck disable=2016
echo "a" && echo "b"
```
You are trying to insert a `shellcheck` directive into an effective single line of code. eg:
```
echo "a" && # shellcheck disable=2016 echo "b"
```
Which, as you can see, is simply not going to work.

**Comment by TinCanTech** [comment]
`shellcheck` analyses the line as it sees it from the shell's perspective.

In your first case, the comment is a directive instructing `shellcheck` how to behave but you are not using `shellcheck` directives correctly. `shellcheck` cannot apply a directive to a partial part of a pipeline.

In your second, the comment is simply a comment with no additional effects.
